cfme/fixtures/utility_vm.py

```diff
@@ -0,0 +1,52 @@
+# -*- coding: utf-8 -*-
+import os.path
+
+import fauxfactory
+import pytest
+
+from cfme.utils.conf import cfme_data
+from cfme.utils.generators import random_vm_name
+from cfme.utils.log import logger
+from cfme.utils.net import wait_pingable
+from cfme.utils.virtual_machines import deploy_template
+from cfme.utils.wait import TimedOutError
+
+
+@pytest.fixture(scope='module')
+def utility_vm():
+    """ Deploy an utility vm for tests to use.
+
+    This fixture creates a vm on provider and then receives its ip.
+    After the test run vm is deleted from provider.
+    """
+    try:
+        root_password = fauxfactory.gen_alphanumeric(length=10)
```

Comment:
_Optional_ thoughts on the root password and root user being stored in a `Credential` object? https://github.com/ManageIQ/integration_tests/blob/d8d4d51abcd51f25ab66d506eff7f7443bae1bed/cfme/base/credential.py#L53

Reply:
Could be done. I don't have much against it. It would actually make possible to log-in to the utility machine without the access to the private key paired with the public key injected to it.

Downside is less security, but I don't think we do need to care about security on private network that much at the moment.

Reply:
Other downside is that now the tests are failing because now they require an entry in the yamls. But I think this is DONE now.